sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/HDFSBackedStateStoreProvider.scala

```diff
@@ -340,13 +341,15 @@ private[state] class HDFSBackedStateStoreProvider(
 
   private def updateFromDeltaFile(version: Long, map: MapType): Unit = {
     val fileToRead = deltaFile(version)
-    if (!fs.exists(fileToRead)) {
-      throw new IllegalStateException(
-        s"Error reading delta file $fileToRead of $this: $fileToRead does not exist")
-    }
     var input: DataInputStream = null
     try {
-      input = decompressStream(fs.open(fileToRead))
+      try {
```

Comment:
OK, i've serialized that. But I think it should work to just unify the try/catch and the try/finally; all other FS open problems were being handled by the original try/finally clause.